### ISSUE TITLE
sorting lists based on selected values

### ISSUE THREAD

**Issue by wgrhstf** [davidstutz/bootstrap-multiselect#982]
I work on quite large dropdown lists with about 150 to 300 options and some preselected values. This is no problem by means of performance, but it's hard to deselect values when you have to scroll that far. 

Is there any possibility to reorder lists automatically so that preselected values always appear on top?

**Comment by Aidurber** [comment]
This would be a breaking change, and could possibly be a bad user experience since options will be moving around. (But I'm no expert in UX)

I don't think that would be out of scope for this library. I think it would be fairly simple to achieve the same result using the onChange method, sorting the data, then rebinding. Not sure how performant it would be though.

**Comment by wgrhstf** [comment]
I agree absolutely when it would be used as a standard. I hoped for that feature maybe as a configuration option. 
For very large lists I would consider it to be an UX advantage not to have to scroll past the bottom of the visible page to find preselected values. When there are preselected values most users want to deselect those values, or add their own selection. I think in both cases it could be an advantage having a possibility to reorder the values. 

As for the onChange idea: I was not so much hoping for a reordering if someone chooses a value (for that would be a bad user experience) but for an initially reordered list based on checked values for easier unchecking them without searching.

**Comment by Aidurber** [comment]
If the issue is the scrolling passed the visible page, have you tried adding a max height?
```javascript
$('#example-with-maxHeight').multiselect({
    maxHeight: 200
});
```

**Comment by wgrhstf** [comment]
No, sorry. The scrolling past the visible page is not an issue in my case. I just used it as an example. The issue is just that I sometimes have to search the list or scroll quite a way before it is possible to unselect a value, if this value is on the very bottom. (I use it in an real-time admin-Interface where speed is a factor)

**Comment by JohanGJ** [comment]
I really like this option too. i have a few dropdowns with hundreds of items, and only 10 selected. that's not that userfriendly too.

**Comment by wgrhstf** [comment]
Using option groups for that is an appealing idea that never crossed my mind. I'm not sure if I can order the data in into groups (all the data is rendered via ractive and it's quite complicated) but I'll give it a try .
Thanks for the thought.
